packages/domain-picker/test/suggestion-item.tsx

```diff
@@ -336,114 +120,61 @@ describe( 'check conditional elements render correctly', () => {
-		const testRequiredProps = {
-			domain: 'testdomain.com',
-			cost: '€12.00',
-			railcarId: 'id',
-			isRecommended: false,
-		};
-
-		render(
-			<SuggestionItem { ...testRequiredProps } onSelect={ jest.fn() } onRender={ jest.fn() } />
-		);
+	it( 'should not render the recommendation badge if is given prop isRecommended false', async () => {
+		renderComponent( { cost: '€12.00', isRecommended: false } );
 
-		// use `queryBy` to avoid throwing an error with `getBy`
-		expect( screen.queryByText( /Recommended/i ) ).toBeFalsy();
+		expect( screen.queryByText( /Recommended/i ) ).not.toBeInTheDocument();
 	} );
 } );
 
 describe( 'test that suggested items are rendered correctly based on availability', () => {
 	it( 'should have the disabled UI state when provided an availabilityStatus of unavailable', () => {
-		const testRequiredProps = {
-			domain: 'testdomain.com',
-			cost: '€12.00',
-			railcarId: 'id',
-			isRecommended: true,
-			isUnavailable: true,
-		};
-
-		render(
-			<SuggestionItem { ...testRequiredProps } onSelect={ jest.fn() } onRender={ jest.fn() } />
-		);
+		renderComponent( { cost: '€12.00', isRecommended: true, isUnavailable: true } );
 
 		// we have to test for the domain and the TLD separately because they get split in the component
-		expect( screen.queryByText( /testdomain/i ) ).toBeTruthy();
-		expect( screen.queryAllByText( /.com/i ) ).toBeTruthy();
 
-		expect( screen.queryByText( /Unavailable/i ) ).toBeTruthy();
-		expect( screen.queryByText( /Recommended/i ) ).toBeFalsy();
-		expect( screen.queryByRole( 'button' )?.getAttribute( 'disabled' ) ).not.toBe( null );
+		const [ domain, tld ] = MOCK_PROPS.domain.split( '.' );
+		expect( screen.getByText( new RegExp( domain, 'i' ) ) ).toBeInTheDocument();
+		expect( screen.getByText( new RegExp( tld, 'i' ) ) ).toBeInTheDocument();
+
+		expect( screen.getByText( /Unavailable/i ) ).toBeInTheDocument();
+		expect( screen.queryByText( /Recommended/i ) ).not.toBeInTheDocument();
+		expect( screen.queryByRole( 'button' ) ).toBeDisabled();
 	} );
 
 	it( 'should have the enabled UI state when provided an availabilityStatus that is available', () => {
-		const testRequiredProps = {
-			domain: 'testdomain.com',
-			cost: '€12.00',
-			railcarId: 'id',
-			isRecommended: true,
-			isUnavailable: false,
-		};
-
-		render(
-			<SuggestionItem { ...testRequiredProps } onSelect={ jest.fn() } onRender={ jest.fn() } />
-		);
+		renderComponent( { cost: '€12.00', isRecommended: true, isUnavailable: false } );
 
 		// we have to test for the domain and the TLD separately because they get split in the component
-		expect( screen.queryByText( /testdomain/i ) ).toBeTruthy();
-		expect( screen.queryAllByText( /.com/i ) ).toBeTruthy();
+		const [ domain, tld ] = MOCK_PROPS.domain.split( '.' );
+
+		expect( screen.getByText( new RegExp( domain, 'i' ) ) ).toBeInTheDocument();
+		expect( screen.getByText( new RegExp( `${ tld }$`, 'i' ) ) ).toBeInTheDocument();
```

Comment:
Nit: compared to previous test:

- There's an empty line before the first `expect`
- The `tld` check also checks for the "end of the expression" (`$`) — this check is not present in the previous test

Reply:
Good one. I've added the "end of the expression" (`$`) since "Re**com**mended" matches too. I agree that it's better to be more specific and concise so I've added the empty line and the `$`.